js/tests/test_examples.test.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * E2E Tests for plugin demos and examples.
+ */
+
+const { execFile } = require('child_process');
+const path = require('path');
+const { promisify } = require('util');
+const execFileAsync = promisify(execFile);
+
+// Paths
+const PLUGINS = path.resolve(__dirname, '../plugins');
+const EXAMPLES_PATH = path.resolve(__dirname, '../examples');
+
+// Require env vars
+const { COMPOSIO_API_KEY, OPENAI_API_KEY } = process.env;
+
+// Plugin test definitions
+const EXAMPLES = {
+  portfolio_generator: {
+    file: path.join(EXAMPLES_PATH, 'portfolio-generator', 'demo.mjs'),
+    match: {
+      type: 'stdout',
+      values: ['🎉Output from agent: '],
+    },
+    env: { COMPOSIO_API_KEY },
+  },
+  lead_generator_agent: {
+    file: path.join(EXAMPLES_PATH, 'reddit-research', 'demo.mjs'),
+    match: {
+      type: 'stdout',
+      values: ['🎉Output from agent: '],
+    },
+    env: { COMPOSIO_API_KEY },
+  },
+  lead_outreach_agent: {
+    file: path.join(EXAMPLES_PATH, 'lead_outreach_agent', 'demo.mjs'),
+    match: {
+      type: 'stdout',
+      values: ['🎉Output from agent: '],
+    },
+    env: { COMPOSIO_API_KEY, OPENAI_API_KEY },
+  }
+};
+
+describe('E2E Tests for plugin demos and examples', () => {
+  jest.setTimeout(180000); // Set timeout to 3 minutes
+
+  for (const [exampleName, example] of Object.entries(EXAMPLES)) {
+    test(`should run ${exampleName} example successfully`, async () => {
+
+      for (const [key, val] of Object.entries(example.env)) {
+        expect(val).not.toBeNull();
+      }
+
+      const options = {
+        env: { ...process.env, ...example.env },
+        cwd: example.cwd || process.cwd(),
+      };
+
+      try {
+        const { stdout, stderr } = await execFileAsync('node', [example.file], options);
+        const output = example.match.type === 'stdout' ? stdout : stderr;
+        for (const match of example.match.values) {
+          expect(output).toContain(match);
+        }
+      } catch (error) {
```

Comment:
Yes, I've used try catch block here because of ```await execFileAsync('node', [example.file], options);```
If this fails, error would be catched by the block

Reply:
Alright makes sense, fixing it